pkg/domain/infra/tunnel/images.go

```diff
@@ -367,3 +363,24 @@ func (ir *ImageEngine) Shutdown(_ context.Context) {
 func (ir *ImageEngine) Sign(ctx context.Context, names []string, options entities.SignOptions) (*entities.SignReport, error) {
 	return nil, errors.New("not implemented yet")
 }
+
+func (ir *ImageEngine) Scp(ctx context.Context, hostsAndImages []string, parentFlags []string, quiet bool) error {
```

Comment:
Using an array for source and destination seems smelly. Why not use parameters for source and destination?